test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateSaveResultTests.cs

```diff
@@ -0,0 +1,288 @@
+            var car2 = new Car { ID = 1002 };
+
+
+            context.AttachTo("Persons", person);
+            context.AttachTo("Cars", car1);
+            context.AttachTo("Cars", car2);
+            context.AddLink(person, "Cars", car1);
+            context.AddLink(person, "Cars", car2);
+
+            var requestInfo = new RequestInfo(context);
+            var serializer = new Serializer(requestInfo);
+            var headers = new HeaderCollection();
+            var clientModel = new ClientEdmModel(ODataProtocolVersion.V4);
+            var entityDescriptor = new EntityDescriptor(clientModel);
+            entityDescriptor.State = EntityStates.Added;
+            entityDescriptor.EntitySetName = "People";
+            entityDescriptor.Entity = person;
+            entityDescriptor.EditLink = new Uri("http://www.foo.com/custom");
+            var requestMessageArgs = new BuildingRequestEventArgs("PATCH", new Uri("http://www.foo.com/Northwind/$metadata#People/$delta"), headers, entityDescriptor, HttpStack.Auto);
+            var odataRequestMessageWrapper = ODataRequestMessageWrapper.CreateRequestMessageWrapper(requestMessageArgs, requestInfo);
+            entityDescriptor.AddRelatedDescriptor(new LinkDescriptor(person, "Cars", car1, clientModel));
+            entityDescriptor.AddRelatedDescriptor(new LinkDescriptor(person, "Cars", car2, clientModel));
+            List<Descriptor> entries = new List<Descriptor> { entityDescriptor };
+            using (ODataMessageWriter messageWriter = Serializer.CreateDeltaMessageWriter(odataRequestMessageWrapper, requestInfo, false /*isParameterPayload*/))
+            {
+                ODataWriterWrapper entryWriter = ODataWriterWrapper.CreateForDeltaFeed(messageWriter, requestInfo.Configurations.RequestPipeline, odataRequestMessageWrapper, requestInfo);
+                serializer.WriteFeedEntry(entries, entryWriter);
+            }
+
+            MemoryStream stream = (MemoryStream)(odataRequestMessageWrapper.CachedRequestStream.Stream);
+            stream.Position = 0;
+
+            string payload = (new StreamReader(stream)).ReadToEnd();
+            payload.Should().Be("{\"@context\":\"http://www.odata.org/service.svc/$metadata#People/$delta\",\"value\":[{\"ID\":100,\"Name\":\"Bing\",\"Cars@delta\":[{\"@id\":\"http://www.odata.org/service.svc/Cars(1001)\",\"ID\":1001},{\"@id\":\"http://www.odata.org/service.svc/Cars(1002)\",\"ID\":1002}]}]}");
+
+        }
+
+        [Fact]
+        public void SerealizeADeltaEntry_With_ADeleteLink()
+        {
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var car1 = new Car { ID = 1001 };
+            var car2 = new Car { ID = 1002 };
+
+            context.AttachTo("Persons", person);
+            context.AttachTo("Cars", car1);
+            context.AttachTo("Cars", car2);
+            context.DeleteLink(person, "Cars", car2);

```

Comment:
You're deleting the link between `person` and `car2`. Where was that link created in the first place?

Reply:
I don't think it is even possible to check whether the link exists between the two from the client side.